Orange/widgets/visualize/owscatterplot.py

```diff
@@ -380,10 +380,16 @@ def update_graph(self, attributes=None, reset_view=True, **_):
         self.graph.zoomStack = []
         if attributes and len(attributes) == 2:
             self.attr_x, self.attr_y = attributes
+        X = BasicStats(self.data[:, self.attr_x], self.attr_x)
+        Y = BasicStats(self.data[:, self.attr_y], self.attr_y)
```

Comment:
But it makes sense to see that the variables contains NaN for each graph update. However, I tried it and it isn't slow at all. The complexity remains `O(m)` where m = no of tuples in the table.